internal/xds/rbac/rbac_engine_test.go

```diff
@@ -1748,8 +1751,8 @@ func (s) TestChainEngine(t *testing.T) {
 					// information to represent incoming RPC's. This will be how a
 					// user uses this API. A user will have to put MD, PeerInfo, and
 					// the connection the RPC is sent on in the context.
-					ctx := metadata.NewIncomingContext(context.Background(), data.rpcData.md)
-
+					ctx, cancel := context.WithTimeout(metadata.NewIncomingContext(context.Background(), data.rpcData.md), defaultTestTimeout)
```

Comment:
This buries the important part of the test where we set the metadata and slightly hurts readability.

Would you mind moving the timeout a bit higher before the comment, at L1750?

Reply:
Moved the context creation above the comment.